src/llm/io_processing/output_parser.cpp

```diff
@@ -24,22 +27,118 @@
 #include "qwen3/reasoning_parser.hpp"
 
 namespace ovms {
+OutputParser::TagLookupStatus OutputParser::StreamOutputCache::lookupTag(const std::string& tag) const {
+    // TODO: Remove debug print
+    std:: cout << "Looking up tag: '" << tag << "' in buffer: '" << buffer << "'" << std::endl;
+
+    if (tag.empty()) {
+        return TagLookupStatus::NOT_FOUND;
+    }
+
+    if (tag.size() > buffer.size()) {
+        /* 
+        If the tag is longer than the buffer, we check if the buffer and tag overlap (either partially or fully for exact match)
+        They do overlap, we assume that tag may appear in the future, so we return FOUND_INCOMPLETE
+        otherwise we return NOT_FOUND
+        */
+        if (stringsOverlap(buffer, tag)) {
+            return TagLookupStatus::FOUND_INCOMPLETE;
+        } else {
+            return TagLookupStatus::NOT_FOUND;
+        }
+    } else if (tag.size() < buffer.size()) {
+        /* 
+        If the tag is shorter than the buffer, we check:
+           a) if the tag is a substring of the buffer (tag is fully matched)
+           b) if the buffer and tag overlap (part of the tag is matched)
+        in the first case we return FOUND_COMPLETE, in the second FOUND_INCOMPLETE
+        otherwise we return NOT_FOUND
+        */
+        if (buffer.find(tag) != std::string::npos) {
+            return TagLookupStatus::FOUND_COMPLETE;
+        } else if (stringsOverlap(buffer, tag)) {
+            return TagLookupStatus::FOUND_INCOMPLETE;
+        } else {
+            return TagLookupStatus::NOT_FOUND;
+        }
+    } else {
+        /*
+        If the tag and buffer are of the same length, we check:
+              a) if they are equal (tag is fully matched)
+              b) if they overlap (part of the tag is matched)
+        in the first case we return FOUND_COMPLETE, in the second FOUND_INCOMPLETE
+        otherwise we return NOT_FOUND
+        */
+        if (buffer == tag) {
+            return TagLookupStatus::FOUND_COMPLETE;
+        } else if (stringsOverlap(buffer, tag)) {
+            return TagLookupStatus::FOUND_INCOMPLETE;
+        } else {
+            return TagLookupStatus::NOT_FOUND;
+        }
+    }
+}
 
-rapidjson::Document OutputParser::parseContentChunk(const std::string& chunk) {
+OutputParser::TagLookupStatus OutputParser::StreamOutputCache::lookupTags(const std::unordered_set<std::string>& tags) const {
+    TagLookupStatus tagLookupStatus = TagLookupStatus::NOT_FOUND;
```

Comment:
This will make result status depend only on last result if there is no complete match. Is it expected?
If we are mostly interested in only FOUND_COMPLETE then we may be wasting time to check for partial completions with set of tags.